js/src/ui/TxList/TxRow/txRow.js

```diff
@@ -137,11 +149,113 @@ class TxRow extends Component {
     return (
       <td className={ styles.timestamp }>
         <div>{ blockNumber && block ? moment(block.timestamp).fromNow() : null }</div>
-        <div>{ blockNumber ? _blockNumber.toFormat() : 'Pending' }</div>
+        <div>{ blockNumber ? _blockNumber.toFormat() : this.renderCancelToggle() }</div>
       </td>
     );
   }
 
+  renderCancelToggle () {
+    const { canceled, editing, isCancelOpen, isEditOpen } = this.state;
+
+    if (canceled) {
+      return (
+        <div className={ styles.pending }>
+          <FormattedMessage
+            id='ui.txList.txRow.canceled'
+            defaultMessage='CANCELED'
+          />
+        </div>
+      );
+    }
+
+    if (editing) {
+      return (
+        <div className={ styles.pending }>
+          <FormattedMessage
+            id='ui.txList.txRow.editing'
+            defaultMessage='EDITING'
```

Comment:
It probably would make more sense having the `defaultMessage` as `editing/Editing` and using text-transform to make it uppercase. (That way we don't need to edit the text and get translations to update when we change the way we want the display the look)

(Same for `SUBMITTING` & `CANCEL`)